python/ray/train/_internal/backend_executor.py

```diff
@@ -474,10 +475,11 @@ def get_with_failure_handling(self, remote_values):
         Returns:
             The resolved objects represented by the passed in ObjectRefs.
         """
-        success = check_for_failure(remote_values)
+        success, exception = check_for_failure(remote_values)
         if success:
             return ray.get(remote_values)
         else:
+            self._last_failure = exception
```

Comment:
aren't we risking memory leaks by setting this as an attribute?

Reply:
How so?

Reply:
According to https://cosmicpercolator.com/2016/01/13/exception-leaks-in-python-2-and-3/, all the temporary variables associated with the traceback will be kept in memory as long as the traceback itself is not gc'd (and it will not be as it is an attribute). I imagine it wouldn't be a huge problem in practice (esp. as we are only dealing with RayActorErrors), but it's something that got me wondering as I vaguely recall that being an issue in some other library a long time ago. Just a musing

Reply:
After reading the article I think it should be fine -- we're only storing one exception, it's stored in a remote actor that is cleaned up eventually, and as it's an ActorError so the actual traceback is limited (as we don't store a serialized exception here). Thanks for bringing this to my attention though!